### PR TITLE
feat: MLIBZ-2533 Make auto-pagination request first if Delta Set and auto-pdagination are enabled

### PR DESCRIPTION
**Pull request by yuliya-guseva**
#### Description
DataStore with AP does not execute Delta Set request

#### Changes
Added auto-pagination request as the first request if delta set and auto-pdagination are enabled

#### Tests
Instrumented

### CODE REVIEW

File: java-api-core/src/com/kinvey/java/store/BaseDataStore.java [hunk 1]
```diff
@@ -409,7 +410,8 @@ public void pushBlocking() throws IOException {
      * should be used with {@link StoreType#SYNC}
      * @param query query to pull the objects
      */
-    public KinveyPullResponse pullBlocking(Query query) throws IOException {
+    @Nonnull
+    public KinveyPullResponse pullBlocking(@Nullable Query query) throws IOException {
         Preconditions.checkArgument(storeType != StoreType.NETWORK, "InvalidDataStoreType");
```
Comment (vinaygahlawat) [inline java-api-core/src/com/kinvey/java/store/BaseDataStore.java]: If we have another `pullBlocking` method that does not take a `Query`, why is the `Query` parameter allowed to be `null`?

File: java-api-core/src/com/kinvey/java/store/BaseDataStore.java [hunk 2]
```diff
@@ -457,104 +460,96 @@ public KinveyPullResponse pullBlocking(Query query, boolean isAutoPagination) th
...
+                executor.shutdown();
+            }
+            response.setListOfExceptions(exceptions);
+            response.setCount(pulledItemCount);
+            if (deltaSetCachingEnabled && lastRequestTime != null) {
+                queryCache.save(new QueryCacheItem(
+                        getCollectionName(),
+                        query.getQueryFilterMap().toString(),
+                        lastRequestTime));
             }
```
Comment (vinaygahlawat) [inline java-api-core/src/com/kinvey/java/store/BaseDataStore.java]: When performing auto-pagination, is this guaranteed to be the earliest `lastRequestTime` of the numerous `GET` requests sent?

File: java-api-core/src/com/kinvey/java/store/BaseDataStore.java (hunk 1 above)
Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/store/BaseDataStore.java]: All `pullBclocking` methods have a `Query` in the parameters.

File: java-api-core/src/com/kinvey/java/store/BaseDataStore.java (hunk 2 above)
Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/store/BaseDataStore.java]: It will be the time from the last request which was executed.

Comment (vinaygahlawat) [reply java-api-core/src/com/kinvey/java/store/BaseDataStore.java]: Based on the design doc, we should set the `lastRequestTime` to the request time returned from the `_count` request made prior to the auto-paginated GET requests:

https://github.com/Kinvey/java-library/blob/072abd9876591a999f036fd66367069c6912a02c/java-api-core/src/com/kinvey/java/store/BaseDataStore.java#L487

**Pull request merged**